lib/src/run.dart

```diff
@@ -144,3 +152,56 @@ stderr:
 
 $stderr""";
 }
+
+
+/// Arguments passed to [Process.run] .
+/// See [Process.run] for more details.
+class RunOptions {
+  final String workingDirectory;
+  final Map<String, String> environment;
+  final bool includeParentEnvironment;
+  final bool runInShell;
+  final Encoding stdoutEncoding;
+  final Encoding stderrEncoding;
+
+  RunOptions({this.workingDirectory, this.environment,
+      this.includeParentEnvironment: true, this.runInShell: false,
+      this.stdoutEncoding: SYSTEM_ENCODING,
+      this.stderrEncoding: SYSTEM_ENCODING});
+
+  /// Create a clone when it's necessary to modify the passed runOptions to
+  /// avoid modifying the argument.
+  RunOptions clone() {
```

Comment:
Done. I stuck with `clone` because I think it`s closest to what it does. `cloneAndUpdate` is closer but that seems cumbersome.